centaur/src/main/scala/centaur/test/ObjectCounter.scala

```diff
@@ -0,0 +1,51 @@
+package centaur.test
+
+import com.amazonaws.services.s3.AmazonS3
+import com.google.cloud.storage.{Blob, Storage}
+import com.google.cloud.storage.Storage.BlobListOption
+import scala.collection.JavaConverters._
+import scala.language.implicitConversions
+
+trait ObjectCounter[A] {
+  def parsePath(regex: String): String => Path = { fullPath =>
+    if (fullPath.matches(regex)) {
+      val prefixLength = 5
+      val bucketAndDashes = fullPath.drop(prefixLength).split("/", 2)
+      val bucket = bucketAndDashes.head
+      val directory = bucketAndDashes.tail.mkString("/")
```

Comment:
Since we are now using `split(..., 2)`, there is no need to use `tail.mkString("/")`